packages/components/src/custom-select-control-v2/custom-select.tsx

```diff
@@ -113,13 +113,20 @@ function _CustomSelect(
 	return (
 		// Where should `restProps` be forwarded to?
 		<div className={ className }>
-			{ hideLabelFromVision ? ( // TODO: Replace with BaseControl
-				<VisuallyHidden as="label">{ label }</VisuallyHidden>
-			) : (
-				<Styled.SelectLabel store={ store }>
-					{ label }
-				</Styled.SelectLabel>
-			) }
+			<Ariakit.SelectLabel
+				store={ store }
+				render={
+					hideLabelFromVision ? (
+						// @ts-expect-error `children` are passed via the render prop
+						<VisuallyHidden />
+					) : (
+						// @ts-expect-error `children` are passed via the render prop
+						<BaseControl.VisualLabel as="div" />
```

Comment:
Using the `label` HTML element is not correct in this case, as also confirmed by the [ariakit docs](https://github.com/ariakit/ariakit/blob/main/packages/ariakit-react-core/src/select/select-label.tsx#L19-L22):

> Since it's not a native select element, we can't use the native label element.

Reply:
Interesting! I played around and confirmed that although a native `label` will work as expected in Chrome, it behaves differently in Firefox and Safari. Clicking on the label not only focuses but clicks the "select button", expanding the dropdown. So yeah, I guess it does need special handling 🫤

Reply:
Yup, that's part of what `Ariakit.SelectLabel` does under the hood